google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -276,7 +275,13 @@ public void processReceivedMessages(List<com.google.pubsub.v1.ReceivedMessage> r
     logger.log(
         Level.FINER, "Received {0} messages at {1}", new Object[] {responseMessages.size(), now});
 
+    try {
+      flowController.reserve(responseMessages.size(), totalByteCount);
+    } catch (FlowController.FlowControlException unexpectedException) {
+      throw new IllegalStateException("Flow control unexpected exception", unexpectedException);
```

Comment:
Renamed.

Dug through some history. FlowController used to have its parameters split: the numerics (how many messages and bytes) and whether to throw exception or stall. The user used to only be able to control the first part. We unconditionally tell the FlowController to stall and never throws. We can therefore "never" see this exception.

Now that we have changed FlowControlSettings to also contain the LimitExceededBehavior, this is no longer true. I guess we can check at construction-time that the behavior is always IGNORE or BLOCK...? WDYT?